File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/opengl/shaders/uniforms/Vec2iUniform.java
package org.saar.lwjgl.opengl.shaders.uniforms;

import org.joml.Vector2ic;
import org.lwjgl.opengl.GL20;

public abstract class Vec2iUniform extends UniformBase implements Uniform {

    @Override
    public final void doLoad() {
        final Vector2ic v = getUniformValue();
        GL20.glUniform2i(getLocation(), v.x(), v.y());
    }

    public abstract Vector2ic getUniformValue();
}

File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/opengl/shaders/uniforms/Vec2iUniform.kt
package org.saar.lwjgl.opengl.shaders.uniforms

import org.joml.Vector2ic
import org.lwjgl.opengl.GL20

abstract class Vec2iUniform : Uniform {

    final override fun load(location: Int) {
        val value: Vector2ic = this.value
        GL20.glUniform2i(location, value.x(), value.y())
    }

    abstract val value: Vector2ic
}